libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -889,7 +898,15 @@ MAV_MISSION_RESULT AP_Mission::mavlink_int_to_mission_cmd(const mavlink_mission_
         cmd.content.location.alt = packet.z * 100.0f;       // convert packet's alt (m) to cmd alt (cm)
 
         switch (packet.frame) {
-
+#if APM_BUILD_TYPE(APM_BUILD_ArduCopter)
+        case MAV_FRAME_LOCAL_NED:
+            if (local_frame_supported) {
+                cmd.content.location.flags.local_frame = 1;
```

Comment:
You have to set origin_alt too. Also, if this is NED, you should change the alt sign (I know the comment says local_frame is alt down, but I see no reason to do that).

Reply:
I understand how it comes in, but we store information in global NEU, what's the reason to not do it for local?

Reply:
I got it. Thank you